images/fcos-bastion-image/root/usr/bin/clean-registries.sh

```diff
@@ -3,9 +3,9 @@ set -x
 
 mapfile -t ports < <(systemctl list-units 'registry@*' --no-pager --quiet | awk -F'[@.]' '{print $2}')
 for port in "${ports[@]}"; do
-  disk_use=$(df /opt/registry-"${port}" --output='pcent' | grep -o '[0-9]*')
+  disk_use=$(df /opt/mirror-registry/registry-"${port}" --output='pcent' | grep -o '[0-9]*')
```

Comment:
?

Reply:
The [script](https://gitlab.cee.redhat.com/aosqe/arm-productivity/-/blob/master/bare-metal/registry/templates/scripts/clean-registry.sh?ref_type=heads) used a: tgt_dir="/opt/mirror-registry"

Therefore, my script is trying to delete wrong path

Reply:
https://github.com/openshift-qe/baremetal-qe-infra/blob/master/images/fcos-bastion-image/root/usr/lib/systemd/system/registry%40.service

the path is correct, what we miss is the garbage collection.

I do not see the garbage collection here:
```
        sed -i -e '/readonly/{n;s/enabled:.*/enabled: true/;}' "${tgt_dir}/registry-${port}/config.yml"
        docker restart poc-registry-${port}
        docker exec poc-registry-${port} /bin/registry garbage-collect /etc/docker/registry/config.yml
        sed -i -e '/readonly/{n;s/enabled:.*/enabled: false/;}' "${tgt_dir}/registry-${port}/config.yml"
        docker restart poc-registry-${port}

```

Reply:
My modified version of the above:
rm -rf /opt/registry-"${port}"/data/docker/registry/v2/repositories/*
    sed -i -e '/readonly/{n;s/enabled:.*/enabled: true/;}' /opt/registry-${port}/config.yml
    systemctl restart registry@"${port}".service
        sed -i -e '/readonly/{n;s/enabled:.*/enabled: false/;}' /opt/registry-${port}/config.yml
    systemctl restart registry@"${port}".service

but still not sure what would be the replacement of:
docker exec poc-registry-${port} /bin/registry garbage-collect /etc/docker/registry/config.yml

Reply:
Actually it should be:
podman exec registry-${port} /bin/registry garbage-collect /etc/docker/registry/config.yml

Reply:
All commands in the script worked, except:
podman exec registry-"${port}" /bin/registry garbage-collect /etc/docker/registry/config.yaml. see below error

Also /etc/docker/registry/config.yaml does not exist actually. it seems mount ddi not work though containers are runnin:
sudo podman ps
CONTAINER ID  IMAGE                          COMMAND               CREATED        STATUS        PORTS                   NAMES
ac5b11a770e2  quay.io/libpod/registry:2.8.2  /etc/docker/regis...  7 minutes ago  Up 7 minutes  0.0.0.0:5000->5000/tcp  registry-5000
4573846e1fa3  quay.io/libpod/registry:2.8.2  /etc/docker/regis...  7 minutes ago  Up 7 minutes  0.0.0.0:6001->6001/tcp  registry-6001
8c13c594f1a9  quay.io/libpod/registry:2.8.2  /etc/docker/regis...  7 minutes ago  Up 7 minutes  0.0.0.0:6002->6002/tcp  registry-6002


+ mapfile -t ports
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4619]: ++ awk '-F[@.]' '{print $2}'
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4618]: ++ systemctl list-units 'registry@*' --no-pager --quiet
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4616]: + for port in "${ports[@]}"
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4621]: ++ df /opt/registry-5000 --output=pcent
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4622]: ++ grep -o '[0-9]*'
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4616]: + disk_use=21
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4616]: + '[' 21 -gt 11 ']'
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4616]: + rm -rf '/opt/registry-5000/data/docker/registry/v2/repositories/*'
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4616]: + sed -i -e '/readonly/{n;s/enabled:.*/enabled: true/;}' /opt/registry-5000/config.yaml
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4616]: + systemctl restart registry@5000.service
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4616]: + podman exec registry-5000 /bin/registry garbage-collect /etc/docker/registry/config.yaml
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4637]: Error: no container with name or ID "registry-5000" found: no such container
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4616]: + sed -i -e '/readonly/{n;s/enabled:.*/enabled: false/;}' /opt/registry-5000/config.yaml
Feb 27 16:15:14 dhcp-1-235-146.arm.eng.rdu2.redhat.com clean-registries.sh[4616]: + systemctl restart registry@5000.service

Reply:
By Adding a sleep of 5 sec after systemctl restart registry@"${port}".service, no complains about container not found

Reply:
Ok, so once the directories in the path exists, the output looks good:
for port in "${ports[@]}"
Feb 27 18:19:43 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[4161]: ++ df /opt/registry-6002 --output=pcent
Feb 27 18:19:43 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[4162]: ++ grep -o '[0-9]*'
Feb 27 18:19:43 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + disk_use=21
Feb 27 18:19:43 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + '[' 21 -gt 11 ']'
Feb 27 18:19:43 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + rm -rf '/opt/registry-6002/data/docker/registry/v2/repositories/*'
Feb 27 18:19:43 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + sed -i -e '/readonly/{n;s/enabled:.*/enabled: true/;}' /opt/registry-6002/config.yaml
Feb 27 18:19:43 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + systemctl restart registry@6002.service
Feb 27 18:19:44 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + sleep 5
Feb 27 18:19:49 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + podman exec registry-6002 /bin/registry garbage-collect /etc/docker/registry/config.yml
Feb 27 18:19:49 dhcp-1-235-235.arm.eng.rdu2.redhat.com podman[4411]: 2024-02-27 18:19:49.226299968 +0000 UTC m=+0.051563717 container exec 4c7297a38c1513f8404e2d2ed6beca33810804e443c06b1e7f8b5174252de54c (image=quay.io/libpod/registry:2.8.2, name=registry-6002, PODMAN_SYSTEMD_UNIT=registry@6002.service)
Feb 27 18:19:49 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[4411]: 0 blobs marked, 0 blobs and 0 manifests eligible for deletion
Feb 27 18:19:49 dhcp-1-235-235.arm.eng.rdu2.redhat.com podman[4411]: 2024-02-27 18:19:49.262250357 +0000 UTC m=+0.087514105 container exec_died 4c7297a38c1513f8404e2d2ed6beca33810804e443c06b1e7f8b5174252de54c (image=quay.io/libpod/registry:2.8.2, name=registry-6002, PODMAN_SYSTEMD_UNIT=registry@6002.service)
Feb 27 18:19:49 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + sed -i -e '/readonly/{n;s/enabled:.*/enabled: false/;}' /opt/registry-6002/config.yaml
Feb 27 18:19:49 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + systemctl restart registry@6002.service
Feb 27 18:19:49 dhcp-1-235-235.arm.eng.rdu2.redhat.com clean-registries.sh[3564]: + exit 0